src/views/RecordPage.vue

```diff
@@ -50,8 +54,41 @@ import {
   IonFabButton,
   IonList,
   IonItem,
+  modalController,
 } from "@ionic/vue";
 import { add, calendar } from "ionicons/icons";
 import WeekHeader from "@/components/WeekHeader.vue";
 import WeekItem from "@/components/WeekItem.vue";
+import RoutineModal from "./RoutineModal.vue";
+import { appStorage } from "@/utils/storage";
+import { AppData, Routine } from "@/utils/app-data";
+import { STORAGE_KEYS } from "@/utils/constant";
+import { onMounted, ref } from "vue";
+
+const routines = ref([] as Routine[]);
+let appDataRef = {} as AppData;
```

Comment:
Why do we need this?

Reply:
Because routines.values can't be written into indexDB, I have to use a different temp variable to keep tracking the appData value changes.